src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                    IValueNode node = GraphQLUtils.GetFieldNodeForGivenFieldName(parameterNodes, referencedEntity.Item1);
+                    PerformDbInsertOperation(context, node.Value, sqlMetadataProvider, ReferencedEntityMultipleCreateStructure, primaryKeysOfCreatedItem);
+                }
+
+                SourceDefinition currentEntitySourceDefinition = sqlMetadataProvider.GetSourceDefinition(entityName);
+                DatabaseObject entityObject = sqlMetadataProvider.EntityToDatabaseObject[entityName];
+                string entityFullName = entityObject.FullName;
+
+                // Populate the relationship fields values for the current entity.
+                if (currentEntitySourceDefinition.SourceEntityRelationshipMap.TryGetValue(entityName, out RelationshipMetadata? relationshipData) && relationshipData is not null)
+                {
+                    foreach ((string relatedEntityName, List<ForeignKeyDefinition> fkDefinitions) in relationshipData.TargetEntityToFkDefinitionMap)
+                    {
+                        DatabaseObject relatedEntityObject = sqlMetadataProvider.EntityToDatabaseObject[relatedEntityName];
+                        string relatedEntityFullName = relatedEntityObject.FullName;
+                        ForeignKeyDefinition fkDefinition = fkDefinitions[0];
+                        if (string.Equals(fkDefinition.Pair.ReferencingDbTable.FullName, entityFullName, StringComparison.OrdinalIgnoreCase)
+                            && string.Equals(fkDefinition.Pair.ReferencedDbTable.FullName, relatedEntityFullName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            int count = fkDefinition.ReferencingColumns.Count;
+                            for (int i = 0; i < count; i++)
+                            {
+                                string referencingColumnName = fkDefinition.ReferencingColumns[i];
+                                string referencedColumnName = fkDefinition.ReferencedColumns[i];
+
+                                if (multipleCreateStructure.CurrentEntityParams!.ContainsKey(referencingColumnName))
+                                {
+                                    continue;
+                                }
+
+                                if (primaryKeysOfCreatedItem.TryGetValue(relatedEntityName, out Dictionary<string, object?>? relatedEntityPKs)
+                                     && relatedEntityPKs is not null
+                                     && relatedEntityPKs.TryGetValue(referencedColumnName, out object? relatedEntityPKValue)
+                                     && relatedEntityPKValue is not null)
+                                {
+                                    multipleCreateStructure.CurrentEntityParams.Add(referencingColumnName, relatedEntityPKValue);
+                                }
+                                else if (multipleCreateStructure.HigherLevelEntityPKs is not null
```

Comment:
This looks like an entirely different concern. We are looping through the relationships for which current entity is the parent and the referencing entity. In this IF condition, though the current entity is the referencing entity, it is not the parent entity but is the child entity. There can be an additional loop which adds to the `CurrentEntityParams`, all the values from the `HigherLevelEntityPKs`.

Reply:
I'm not sure if I follow your comment but this is along the same lines as your next comment.  (Link: https://github.com/Azure/data-api-builder/pull/1994#discussion_r1559491608)

I'll provide an explanation of the logic, hopefully that answers your question.

The goal is to populate `CurrentEntityParams` with all the fields necessary to successfully build the database query. Once, all the necessary fields are populated, it is then used to construct the `SqlInsertStructure`.

At the very beginning of the flow, `CurrentEntityParams` will be populated with scalar fields belonging to the current entity. In other words, the fields that are marked as required fields in the schema.

The pending fields that are necessary are the relationship fields. The logic then proceeds to identify and categorize relationship inputs as `referenced` or `referenced`. The `referenced` entities will be processed first and then the PKs will be populated in `CurrentEntityParams`. 

But, it can also be the case that the current entity references an additional entity from a higher level. Ex: `Review` references `Book`. I've explained with a concrete example in the next comment https://github.com/Azure/data-api-builder/pull/1994#discussion_r1559491608. I've explained why we need both `CurrentEntityParams` as well as `HigherLevelEntityPKs`. Kindly go through that comment before reading further.

So, essentially, the relationship fields can come from 
a) Entities referenced in the current level.
b) Entities referenced at a higher level. 


The loop iterates through all the relationships,
a) Checks if the foreign key is already provided in the input request. Ex: Using `publisher_id` directly in `createbook`  request. If foreign key is already present, nothing further needs to be done.
b) Check if the foreign key can be derived from `primaryKeysOfCreatedItem`. If so, populate it.
c) Check if the foreign key can be derived from `HigherLevelEntityPKs`. If so, populate it.

If the foreign key is not attainable through any of the above methods and a) is not auto-generated b) does not contain\ default value c) is not nullable, then we simply lack necessary details to form the database query; in which case we throw an exception.

Reply:
Please refer to comment: https://github.com/Azure/data-api-builder/pull/1994#discussion_r1564910261